### PR TITLE
feat: edit vardict and tnscope_umi VCF files for scout upload

### PR DESCRIPTION
**Pull request by ashwini06**
### This PR:

Fixes: #944 

Added: 
- Python script which could edit the INFO field in the VCF files (only for Vardict and TNscope_umi) by adding information about the variant caller and the VCF filepath.
- Python script requires dependencies, `cyvcf2` and `click` is added to the `annotate` container

Changed: 
- To be compatible with scout, the UMI filename `TNscope_umi` is changed to `tnscope_umi`

### Review and tests: 
- [x] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

**Comment by ashwini06** [comment]
Tests: 

New VCF files are located here `/home/proj/stage/cancer/cases/sweetelf_test_APJ/analysis/vep/SNV.somatic.sweetelf_test_APJ.tnscope_umi.all.filtered.pass.vcf.gz`

tnscope_umi_edited_vcf
 `/home/proj/stage/cancer/cases/sweetelf_test_APJ/analysis/vep/SNV.somatic.sweetelf_test_APJ.vardict.all.filtered.pass.vcf.gz`

vardict_edited_vcf

Scout Upload: https://scout-stage.scilifelab.se/cust000/balsamic-validation-gmsmyeloid-tumor-only-2/cancer/e8c28f145af98e97dc871ca811f7bb3e?cancer=yes

Screenshot 2022-06-09 at 13 26 14

File: BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_only.rule [hunk 1]
```diff
@@ -83,23 +93,26 @@ tabix -p vcf -f {output.vcf_pass_tnhaplotyper};
...
+    vcf_pass_TNscope_umi = vep_dir + "{var_type}.somatic.{case_name}.tnscope_umi.all.filtered.pass.vcf.gz",
+    bcftools_counts = vep_dir + "{var_type}.somatic.{case_name}.tnscope_umi.all.filtered.pass.stats"
   benchmark:
     Path(benchmark_dir, 'bcftools_filter_TNscope_umi_tumor_only' + "{var_type}.somatic.{case_name}.tsv").as_posix()
   singularity:
-    Path(singularity_image, config["bioinfo_tools"].get("bcftools") + ".sif").as_posix()
+    Path(singularity_image, config["bioinfo_tools"].get("ensembl-vep") + ".sif").as_posix()
   params:
     pop_freq = [SENTIEON_CALLER.pop_freq_umi.tag_value, SENTIEON_CALLER.pop_freq_umi.filter_name],
     housekeeper_id = {"id": config["analysis"]["case_id"], "tags": "research"},
```
Comment (ivadym) [inline BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_only.rule:107]: This one should also be clinical, right?

```suggestion
    housekeeper_id = {"id": config["analysis"]["case_id"], "tags": "clinical"},
```

**Review by ivadym — COMMENTED** [review]
I just have some minor comments. Great job with the Scout tests 🥇!

File: BALSAMIC/assets/scripts/edit_vcf_info.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# vim: syntax=python tabstop=4 expandtab
+# coding: utf-8
+
+""" Script to include variant caller information to the VCF info field and output to new VCF file """
+
+import click
+import gzip
+from cyvcf2 import VCF, Writer
```
[the remark this answers is not in the document]
Comment (ashwini06) [reply BALSAMIC/assets/scripts/edit_vcf_info.py:8]: Thats a good idea to have common standard packages for VCF parsing. cyvcf2 has simple built-in-functions to modify VCF files, so is the vcfpy I think. Need to invest some time to find the commonality and generalize these functions within BALSAMIC. But due to my restricted timeline, I would rather spend time on the other pending tasks of mine :)

File: BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_only.rule (hunk 1 above)
Comment (ashwini06) [reply BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_only.rule:107]: Done it for TN, forgot to change this ;)

File: BALSAMIC/assets/scripts/edit_vcf_info.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ivadym) [reply BALSAMIC/assets/scripts/edit_vcf_info.py:8]: I agree, its not a priority now :)

**Pull request merged**